gematria/datasets/annotating_importer.h

```diff
@@ -0,0 +1,120 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Contains support code for importing and annotating basic block data.
+
+#ifndef THIRD_PARTY_GEMATRIA_GEMATRIA_DATASETS_ANNOTATING_IMPORTER_H_
+#define THIRD_PARTY_GEMATRIA_GEMATRIA_DATASETS_ANNOTATING_IMPORTER_H_
+
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+#include "absl/status/statusor.h"
+#include "gematria/datasets/bhive_importer.h"
+#include "gematria/llvm/canonicalizer.h"
+#include "gematria/llvm/disassembler.h"
+#include "llvm/Object/Binary.h"
+#include "llvm/Object/ELFObjectFile.h"
+#include "quipper/perf_parser.h"
+#include "quipper/perf_reader.h"
+
+namespace gematria {
+
+// Importer for annotated basic blocks.
+class AnnotatingImporter {
+ public:
+  // Creates a new annotation collector from a given canonicalizer. The
+  // canonicalizer must be for the architecture/microarchitecture of the data
+  // set. Does not take ownership of the canonicalizer.
+  explicit AnnotatingImporter(const Canonicalizer* canonicalizer);
+
+  // Reads an ELF object along with a corresponding `perf.data`-like file and
+  // returns a vector of annotated `BasicBlockProto`s consisting of basic blocks
+  // from the ELF object annotated using samples from the `perf.data`-like file.
+  absl::StatusOr<std::vector<BasicBlockWithThroughputProto>>
+  GetAnnotatedBasicBlockProtos(const std::string_view elf_file_name,
```

Comment:
Nit & no action really needed: for arguments passed by value, `const` in a declaration doesn't mean much and the implementation can add or remove it at will (and since the argument is passed by value, whether there is a const or not makes no difference for the caller).